lib/mayaUsd/render/vp2RenderDelegate/basisCurves.cpp

```diff
@@ -1193,28 +1192,26 @@ HdVP2BasisCurves::_UpdateDrawItem(
         }
     }
 
-    if (itemDirtyBits & HdChangeTracker::DirtyVisibility) {
-        drawItemData._enabled = drawItem->GetVisible();
-        stateToCommit._enabled = &drawItemData._enabled;
-    }
+    // Determine if the render item should be enabled or not.
+    if (itemDirtyBits & (HdChangeTracker::DirtyVisibility |
+                         HdChangeTracker::DirtyRenderTag |
+                         HdChangeTracker::DirtyPoints |
+                         HdChangeTracker::DirtyExtent |
+                         DirtySelectionHighlight)) {
+        bool enable = drawItem->GetVisible() && !_curvesSharedData._points.empty();
 
-    if (isDedicatedSelectionHighlightItem) {
-        if (itemDirtyBits & DirtySelectionHighlight) {
-            const bool enable =
-                (_selectionState != kUnselected) && drawItem->GetVisible();
-            if (drawItemData._enabled != enable) {
-                drawItemData._enabled = enable;
-                stateToCommit._enabled = &drawItemData._enabled;
-            }
+        if (isDedicatedSelectionHighlightItem) {
+            enable = enable && (_selectionState != kUnselected);
         }
-    }
-    else if (drawMode == MHWRender::MGeometry::kBoundingBox) {
-        if (itemDirtyBits & HdChangeTracker::DirtyExtent) {
-            const bool enable = !range.IsEmpty() && drawItem->GetVisible();
-            if (drawItemData._enabled != enable) {
-                drawItemData._enabled = enable;
-                stateToCommit._enabled = &drawItemData._enabled;
-            }
+        else if (drawMode == MHWRender::MGeometry::kBoundingBox) {
+            enable = enable && !range.IsEmpty();
+        }
+
+        enable = enable && drawScene.DrawRenderTag(_curvesSharedData._renderTag);
+
+        if (drawItemData._enabled != enable) {
+            drawItemData._enabled = enable;
+            stateToCommit._enabled = &drawItemData._enabled;
```

Comment:
Streamline the code about the enable/disable state.

Reply:
@williamkrick The change here will conflict with #558. I think this change will be clearer and streamline all the relevant code about enable/disable state. It also avoids the query of the render tag when there is points change (e.g. deformation). What do you think?

Reply:
Maybe this change also fixes a potential issue that if you switch the purpose of the prim and then switch the viewport mode the render item state won't be updated?

Reply:
Sure lets do it your way. I'll take that part of the changes out of #558. We still need _updateRenderTags because we still need to mark rprims that are becoming hidden or visible dirty. Without marking them dirty we don't call Sync and so can't update.

I'll update #558 to be what I think is correct and we can discuss more.